apps/backend/convex/provider/openrouter/generationData.ts

```diff
@@ -10,7 +10,10 @@ const OpenRouterMetadata = z.object({
   id: z.string(),
   model: z.string(),
   total_cost: z.number(),
-  finish_reason: z.string(),
+  finish_reason: z
+    .string()
+    .nullable()
+    .transform((val) => (val === null ? '[null]' : val)),
```

Comment:
It's important to capture the actual response from the API which we do control and change behaviour at anytime. In this case it has to be a string to avoid a database migration, so representing null in this way is the best solution.